client/ayon_applications/manager.py

```diff
@@ -163,12 +359,188 @@ def launch(self, app_name, **data):
+
+        # Get all processes and check which ones are inactive
+        all_processes = self.get_all_process_info()
+        inactive_hashes = []
+
+        for process in all_processes:
+            if not process.active:
+                process_hash = self.get_process_info_hash(process)
+                inactive_hashes.append(process_hash)
+
+        if not inactive_hashes:
+            return 0
+
+        cursor = self._process_storage.cursor()
+        placeholders = ','.join('?' * len(inactive_hashes))
+        cursor.execute(
+            f"DELETE FROM process_info WHERE hash IN ({placeholders})",
+            inactive_hashes
+        )
+        self._process_storage.commit()
+        return cursor.rowcount
+
+    @staticmethod
+    def _is_process_running_psutils(pid: int) -> bool:
+        """Check if a process is running using psutil.
+
+        Args:
+            pid (int): Process ID to check.
+
+        Returns:
+            bool: True if the process is running, False otherwise.
+
+        """
+        # import check should be done before invoking this method
+        if not psutil:
+            msg = "psutil module is not available."
+            raise RuntimeError(msg)
+
+        return psutil.pid_exists(pid)
+
+    @staticmethod
+    def _are_processes_running(pids: list[int]) -> list[tuple[int, bool]]:
+        """Check if the processes are still running.
+
+        This checks for presence of `psutil` module and uses it if available.
+        If `psutil` is not available, it falls back to using `os.kill` on Unix
+        systems or `tasklist` command on Windows to check if the processes
+        are running. `psutil` is preferred because it is more reliable and
+        provides a consistent interface across platforms. But since it is a
+        not pure Python module, it may not be available on all systems.
+
+        The batch check is done to avoid multiple calls to the system
+        to check for each process individually, which can be inefficient -
+        especially on Windows where `tasklist` can be slow for many processes.
+        `tasklist` supports querying multiple processes at once using
+        the `/FI` filter option.
+
+        We should refactor this method once we find out that the fallback
+        method is not needed anymore.
+
+        Args:
+            pids (list[int]): Processes ID to check.
+
+        Returns:
+            list[tuple[int, bool]]: List of tuples with process ID and
+                boolean indicating if the process is running.
+        """
+        result: list[tuple[int, bool]] = []
+
+        if not pids:
+            return result
+
+        try:
+            import psutil
+            for pid in pids:
+                is_running = psutil.pid_exists(pid)
+                result.append((pid, is_running))
+
+            return result
+
+        except ImportError:
+            # Fallback for systems without psutil
```

Comment:
The code should not live under the `ImportError` handler.
```suggestion
        except ImportError:
            pass

        # Fallback for systems without psutil
```